cmp/form/JsonField.js

```diff
@@ -22,19 +23,49 @@ import 'codemirror/addon/lint/lint.js';
 
 import './JsonField.css';
 
-
 /**
- *   A JSON Editor
- *
- * @prop rest, see general properties for HoistField
+ * A JSON Editor
  *
- * @prop width, width of field, in pixels
- * @prop height, width of field, in pixels
- * @prop lineWrapping, Whether field should scroll or wrap for long lines. Defaults to false
+ * See HoistField for properties additional to those documented below.
  */
 @hoistComponent()
 export class JsonField extends HoistField {
 
+    static propTypes = {
+        /** width of field, in pixels */
+        width: PT.number,
+        /** height of field, in pixels */
+        height: PT.number,
+        /**
+         * Configuration object with any properties supported by the CodeMirror api.
+         * See: https://codemirror.net/doc/manual.html#api_configuration for details.
+         * */
+        editorProps: PT.object
+    };
+
+    static defaultProps = {
```

Comment:
I think we have good reason to use defaultProps here. As mentioned in the previous PR:

Looking at the code mirror source it appears that an undefined height or width will be handled the same as null in the setSize method, but in the docs it suggests null. Including as defaultProps allows us to follow the docs, be more self documenting, and avoids having to use 'let' in the implementation to allow conversation as we were doing before.

https://github.com/codemirror/CodeMirror/blob/2c741bd6742678f67e812307b2f78db4f4566d9d/src/edit/methods.js#L392

cm.setSize(width: number|string, height: number|string)
Programmatically set the size of the editor (overriding the applicable CSS rules). width and height can be either numbers (interpreted as pixels) or CSS units ("100%", for example). You can pass null for either of them to indicate that that dimension should not be changed.